### PR TITLE
Add the channel subsetting functionality

### PR DESCRIPTION
**Pull request by danholdaway**
## Description

Remove channels based on user selection

### CODE REVIEW

File: src/eva/data/ioda_obs_space.py [hunk 1]
```diff
@@ -64,23 +71,53 @@ def execute(self, data_collections):
                 # Loop over filenames
                 for filename in filenames:
 
-                    # Open file and add collection
-                    with xr.open_dataset(filename, group=group) as ds:
+                    # Read header part of the file to get coordinates
```
Comment (CoryMartin-NOAA) [inline src/eva/data/ioda_obs_space.py:74]: Rahul will say to do a with open like you had before :-)

**Review by CoryMartin-NOAA — COMMENTED** [review]
One comment about the `with foo as bar` being the 'preferred' way to open files, otherwise looks good to me.

File: src/eva/data/ioda_obs_space.py (hunk 1 above)
Comment (danholdaway) [reply src/eva/data/ioda_obs_space.py:74]: It leads to substantial indenting, which I felt reduced the readability of the code. There are the associated closing of the DataSets later in the code so I don't think it makes too much difference. But I'm happy to return it if you think it's better?

Comment (CoryMartin-NOAA) [reply src/eva/data/ioda_obs_space.py:74]: Personally, I think it's fine the way it is. But I know that the 'pythonic' way is the 'with ...'. But leave it as is, readability is important.

Comment (danholdaway) [reply src/eva/data/ioda_obs_space.py:74]: OK, will do. We can look again at it. Especially if the code changes as we learn more about the way it works when I start adding filters and trying to make plots.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**